adapters/tlsio_wolfssl.c

```diff
@@ -422,7 +422,7 @@ static int on_io_send(WOLFSSL *ssl, char *buf, int sz, void *context)
         LogError("Failed sending bytes through underlying IO");
         tls_io_instance->tlsio_state = TLSIO_STATE_ERROR;
         indicate_error(tls_io_instance);
-        result = 0;
+        result = WOLFSSL_CBIO_ERR_WANT_WRITE;
```

Comment:
Shouldn't this be `sz` (assuming it is negative).

According to https://www.wolfssl.com/wolfssl-alternative-io-support/ the error code might have a different meaning: 
> If the “cipher-send” buffer isn’t big enough, the callback should return -2 (WOLFSSL_CBIO_ERR_WANT_WRITE).

Reply:
I noticed it was used in that sample for buffer not being the correct size, but did not find a description for the usage of WOLFSSL_CBIO_ERR_WANT_WRITE specifically.  I interpreted it being used there as an example of its usage, but is not limited to that scenario. 

We cannot assume on `xio_send` failure that the size returned is negative.  Size could still be 0 or positive based on `socketio_send`

Reply:
I think the options are either to return a negative number explicitly, like '-1' per the CBIOSend sample (https://github.com/wolfSSL/wolfssl-examples/blob/master/custom-io-callbacks/file-server/file-server.c), or use the WOLFSSL error macro per the send_client sample (https://github.com/wolfSSL/wolfssl-examples/blob/master/custom-io-callbacks/file-server/file-server.c).

Reply:
I think the options are either to return a negative number explicitly, like '-1' per the CBIOSend sample (https://github.com/wolfSSL/wolfssl-examples/blob/master/custom-io-callbacks/file-server/file-server.c), or use the WOLFSSL error macro per the send_client sample (https://github.com/wolfSSL/wolfssl-examples/blob/master/custom-io-callbacks/file-server/file-server.c).

Reply:
https://github.com/wolfSSL/wolfssl-examples/blob/2f048ad8dc0c556f021b614bc48e2dce2596fe18/custom-io-callbacks/file-server/file-server.c#L87-L102

https://github.com/wolfSSL/wolfssl-examples/blob/2f048ad8dc0c556f021b614bc48e2dce2596fe18/embedded/tls-client-server.c#L84-L99

Reply:
WOLFSSL_CBIO_ERR_GENERAL // -1